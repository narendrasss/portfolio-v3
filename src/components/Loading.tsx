import React from 'react';
import styled from 'styled-components';
import Layout from './Layout';

const AnimatedBox = styled.div`
  width: 50%;
  height: 100%;
  background: black;
`;

const Center = styled.h1`
  position: absolute;
  color: white;
`;

const Loading: React.FC = () => (
  <Layout>
    <AnimatedBox />
    <AnimatedBox />
    <Center>hello!</Center>
  </Layout>
);

export default Loading;
