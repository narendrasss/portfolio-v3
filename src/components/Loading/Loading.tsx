import React from 'react';
import styled, { css } from 'styled-components';
import Heading from '../base/Heading';
import Main from '../Main/Main';
import { initStyle, keyframe } from '../../animations/fade/fadeInOut';

export interface LoadingProps {
  duration?: number;
}

const Message = styled(Heading)<LoadingProps>`
  ${({ duration = 1.2 }) => css`
    ${initStyle}
    animation: ${keyframe} ${duration}s ease-in-out;
  `}
`;

const Loading: React.FC<LoadingProps> = ({ duration }) => (
  <Main alignItems="center" justifyContent="center">
    <Message duration={duration}>hello!</Message>
  </Main>
);

export default Loading;
