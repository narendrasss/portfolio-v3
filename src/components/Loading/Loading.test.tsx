import React from 'react';
import { render } from 'react-testing-library';
import Loading from './Loading';

describe('Loading', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
});
