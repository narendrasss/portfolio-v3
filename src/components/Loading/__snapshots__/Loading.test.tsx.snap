// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading renders without crashing 1`] = `
<DocumentFragment>
  <main
    class="sc-bdVaJa sc-bwzfXH sc-htoDjs eUrBiN"
  >
    <h1
      class="sc-bdVaJa sc-htpNat sc-dnqmqq hclutY sc-bxivhb hyqNPW"
      font-size="3"
      font-weight="bold"
    >
      hello!
    </h1>
  </main>
</DocumentFragment>
`;
